nucypher/characters/lawful.py

```diff
@@ -1735,6 +1738,62 @@ def _reencrypt(self, kfrag: KFrag, work_order: 'WorkOrder', alice_verifying_key:
         # ... and finally returns all the re-encrypted bytes
         return cfrag_byte_stream
 
+    def set_recorded_fleet_state(self, fleet_state: ArchivedFleetState):
+        self._recorded_fleet_state = fleet_state
+
+    #
+    # Status metadata
+    #
+
+    def status_info(self, raise_invalid=True, omit_known_nodes=False, is_local=True):
+        # FIXME: is anyone using `raise_invalid=True`? Or is it always `False`?
+
+        try:
+            worker_address = self.worker_address
+        # FIXME: how did such a node end up in `known_nodes`?
+        except self.StampNotSigned:
```

Comment:
> how did such a node end up in `known_nodes`?

Fair point.  This was almost certainly a hot-fix once to keep the centralized monitor running.  Perhaps we can catch a more generic error here to prevent any unintentional crash behavior?

Reply:
I'm starting to think that crashing here is a desired behavior. In fact, we should have crashed much earlier. We got a node here that's invalid yet still present in `known_nodes`. It means it can be selected for granting, or to be a teacher. This is a serious problem, and we can't just sweep it under the rug like that. 

Another consideration is that we need to trust the objects we create. Otherwise, to be consistent, we would have to add `raise_invalid` to pretty much every method call. 

If we were to allow for an exception in `status_info`, this should be handled in `server.py`. I removed the `try` block from `status_info` for now, but I will wait on adding it to the status page endpoint - it creates some API complications, and I'd rather wait until we decide if it is necessary.

Reply:
This raises the question, if not here, where then do we expect this kind of exception to be raised in `server.py` - Is this essentially a side-effect of calling `NodeSprout.mature` on every peer?

Reply:
For this specific exception, I expect the node without a worker address to not appear in `known_nodes` at all and be weeded out somewhere in `remember_node()`. Whether we bucket it, or just ignore it, is details. This exception shouldn't even reach `server.py`.

Reply:
After some investigation it turns out that `FleetSensor` can indeed contain unverified nodes. This may or may not change in #2513. For now I'll just add a verified flag in the returned node status info, and make `worker_address` nullable. `nucypher-monitor` doesn't use `worker_address` anyway, so for it nothing will change (although we may add some visualization of verified/unverified nodes later). Ursula's status page will separate verified and unverified nodes.